backend/src-common/src/main/java/org/eclipse/sw360/datahandler/db/ComponentDatabaseHandler.java

```diff
@@ -619,6 +620,14 @@ public RequestStatus updateComponent(Component component, User user) throws SW36
             if (isComponentNameChanged) {
                 updateComponentDependentFieldsForRelease(component,referenceDocLogList,user.getEmail());
             }
+
+            if (component.getComponentType() != null && !component.getComponentType().equals(actual.getComponentType())) {
+            	String comp_type = component.getComponentType().name();
+            	if (comp_type != "OSS") {
```

Comment:
Replace line `625` and `626` with single line:
`if (!ComponentType.OSS.equals(component.getComponentType())) `

Reply:
done the changes.